crates/client/l1-messages/src/contract.rs

```diff
@@ -32,12 +38,15 @@ pub fn parse_handle_l1_message_transaction(
         .try_into()
         .map_err(L1EventToTransactionError::InvalidNonce)?;
 
-    let calldata: Vec<Felt252Wrapper> = event
-        .payload
-        .iter()
-        .map(|param| Felt252Wrapper::try_from(sp_core::U256(param.0)))
-        .collect::<Result<Vec<Felt252Wrapper>, Felt252WrapperError>>()
-        .map_err(L1EventToTransactionError::InvalidCalldata)?;
+    let mut calldata: Vec<Felt252Wrapper> = vec![from_address];
```

Comment:
you can preallocate the correct length there: event.payload.len() + 1

Reply:
and use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect_into to populate the allocated vec